pgml-dashboard/src/utils/url_parser.rs

```diff
@@ -0,0 +1,128 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+
+lazy_static! {
+    static ref DOMAIN_REGEX: Regex =
+        Regex::new(r"^https?://((localhost:8000)|(.*\.?postgresml\.org))").unwrap();
+    static ref UUID_REGEX: Regex = Regex::new(".{8}-.{4}-.{4}-.{4}-.{12}").unwrap();
+}
+
+pub struct ClusterUrlParser;
+
+impl ClusterUrlParser {
+    // Returns option containing path. If regex does not match, returns None.
+    pub fn get_path(url: &str) -> Option<String> {
```

Comment:
Rocket allows you to get a path without resorting to a regex: https://docs.rs/rocket/latest/rocket/struct.Request.html#method.uri